### ISSUE TITLE
centos zsh can only be upgraded to 5.0.2.

### ISSUE THREAD

**Issue by chandrahas-abburi** [Powerlevel9k/powerlevel9k#1355]
Thanks for opening an issue! For a project that deals with as many different things as P9k, debugging problems can be difficult. Please follow the guide, below, to create a bug report that will help us help you!

### Before Opening a Bug
P9k is lovingly maintained by volunteers, and we are happy to help you! You can help us by first making sure your issue hasn't already been solved before opening a new one. Please check the [Troubleshooting Guide](https://github.com/bhilburn/powerlevel9k/wiki/Troubleshooting) first. Many issues are actually local configuration problems, which may have previously been solved by another user - be sure to also [search the existing issues](https://github.com/bhilburn/powerlevel9k/issues?utf8=%E2%9C%93&q=is%3Aissue) before opening a new one.

Once you've done these things, you can delete this section and proceed `=)`

-----

#### Describe Your Issue
What is happening?

Most issues are best explained with a screenshot. Please share one if you can!

#### Have you tried to debug or fix it?

Have you tinkered with your settings, and what happened when you did? Did you find a bit of code that you think might be the culprit? Let us know what you've done so far!

#### Environment Information

This information will help us understand your configuration.

  - What version of ZSH are you using? You can use `zsh --version` to see this.
  - Do you use a ZSH framework (e.g., Oh-My-ZSH, Antigen)?
  - How did you install P9k (cloning the repo, by tarball, a package from your OS, etc.,)?
  - What version of P9k are you using?
  - Which terminal emulator do you use?

#### Issues with Fonts & Icons
You may delete this section if your issue is not font / icon related.

  - Which font do you use?
  - Which [font configuration mode](https://github.com/bhilburn/powerlevel9k/wiki/About-Fonts) are you using? You can check this with (`echo $POWERLEVEL9K_MODE`).
  - Please share the contents of `$P9k/debug/font-issues.zsh`.
  - If this is an icon problem, does the output of `$ get_icon_names` look correct?

zsh version -zsh 5.0.2 (x86_64-redhat-linux-gnu)
powerlevel 9k gives many git errors
Need a workaround or support for centos 7

**Comment by romkatv** [comment]
> centos zsh can only be upgraded to 5.0.2

This works for me:

```zsh
sudo yum update -y
sudo yum install -y git make ncurses-devel gcc autoconf man
git clone -b zsh-5.7.1 https://github.com/zsh-users/zsh.git /tmp/zsh
cd /tmp/zsh
./Util/preconfig
./configure
sudo make -j 20 install.bin install.modules install.fns
```

`/usr/local/bin/zsh --version` prints `zsh 5.7.1 (x86_64-pc-linux-gnu)`.

Does it work for you?

**Comment by chandrahas-abburi** [comment]
yes, it does. Thank you @romkatv for the quick response

**Comment by romkatv** [comment]
> No, it doesnt.

Could you post the failing command and its output?

> Thank you @romkatv for the quick response. i think i need to make oh my zsh use the 5.7.1 version
> 
> echo $ZSH_VERSION still gives 5.0.2

This makes sense because you still have `/bin/zsh` as your user shell. `/bin/zsh --version` is `5.0.2` while `/usr/local/bin/zsh --version` is `5.7.1`.

> zsh -- version gives 5.7.1

Where did zsh version 5.7.1 come from? Did the build command actually succeed?

> if I type zsh in the command line then the error vanishes.

You can change your default shell to the new version by typing these commands:

```zsh
sudo grep -qF =zsh /etc/shells || sudo tee -a /etc/shells <<<=zsh
chsh -s =zsh
```

**Comment by pingsutw** [comment]
@romkatv awesome, it works for me

**Comment by siva3060** [comment]
> > centos zsh can only be upgraded to 5.0.2
> 
> This works for me:
> 
> ```shell
> sudo yum update -y
> sudo yum install -y git make ncurses-devel gcc autoconf man
> git clone -b zsh-5.7.1 https://github.com/zsh-users/zsh.git /tmp/zsh
> cd /tmp/zsh
> ./Util/preconfig
> ./configure
> sudo make -j 20 install.bin install.modules install.fns
> ```
> 
> `/usr/local/bin/zsh --version` prints `zsh 5.7.1 (x86_64-pc-linux-gnu)`.
> 
> Does it work for you?

**Comment by siva3060** [comment]
I have followed the above procedure able to install the zshell how ever when i try to sourece the .bashrc file it throws error of saying module not installed of zsh was the error it is throwing

**Comment by romkatv** [comment]
@siva3060 What?

**Comment by siva3060** [comment]
i have configure the zsh to my rhel machine but after downloading the on-my-zsh and reloading the zshrc script using source commad it throws errors saying modules of zsh not installed

**Comment by romkatv** [comment]
Aww, that sucks :frowning_face: Sorry to hear it didn't work for you.

**Comment by siva3060** [comment]
is there any possible way to fix this cus i have been on this since last night i dont see any resource thank you

**Comment by romkatv** [comment]
There must be some way but I don't know what it might be, unfortunately. You might have better luck asking for help from people who have something to do with the technologies you are using: rhel, zsh and oh-my-zsh, if I understand correctly.

**Comment by siva3060** [comment]
yes, yum  manager has 5.0.2 and havent received any update later thath

**Comment by romkatv** [comment]
I feel ya.

**Comment by siva3060** [comment]
Tried to install with rpm ending with dependiency erros

**Comment by romkatv** [comment]
> Tried to install with rpm ending with dependiency erros

Shiiiiit, I hate those.
